zlib-ng.h.in

```diff
@@ -1812,9 +1812,9 @@ typedef enum {
 } zng_deflate_param;
 
 typedef struct {
-    zng_deflate_param param;  /* parameter ID */
     void   *buf;              /* parameter value */
     size_t  size;             /* parameter value size */
+    zng_deflate_param param;  /* parameter ID */
     int32_t status;           /* result of the last set/get call */
 } zng_deflate_param_value;
```

Comment:
It would be helpful here to explain the logic in a comment why `param` needs to come after `buf` and `size`. Otherwise someone who is not familiar with structure member alignment might try to revert the change.

Reply:
Actually I think param should be last.
Because, depending on compiler, if you ask it to pack structs, it can end up taking only 1 byte. Not that that would be a good idea. I don't think the order matters at all here, so it might as well go last.